### PR TITLE
Bumped supported versions of Node and TypeScript in CircleCI

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### PR checklist

- [x] Addresses an existing issue: supports #4914 
- ~[ ] New feature, bugfix, or enhancement~
  - ~[ ] Includes tests~
- ~[ ] Documentation update~

#### Overview of change:

Removes the now-EOL Node 6 from CircleCI, and explicitly adds TypeScript versions 3.1-3.8 to tests.

I noticed `appveyor.yml` is still hanging around, so I removed that too.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -118,7 +106,7 @@ jobs:
             - run: yarn test
 
     test2.9:
-        docker: *docker-node-6
+        docker: *docker-node-
```
Comment (adidahiya) [inline .circleci/config.yml]: missing `-8`, which might be tripping up circle in running the workflow

Comment (JoshuaKGoldberg) [reply .circleci/config.yml]: Ha, thanks. I'm surprised Circle didn't give a better error indicator for the invalid reference...

**Comment by JoshuaKGoldberg** [comment]
I think this ranks up there with one of the most amusing open source CircleCI flows I've seen in a JavaScript/Node project...

Screenshot of the TSLint CircleCI flow with 18 parallel end steps

**Comment by JoshuaKGoldberg** [comment]
@adidahiya I'm wondering if we could get away with maybe a third of those languages... say, 3.1, 3.4, and 3.8?

**Pull request merged**